src/lib/emitter.c

```diff
@@ -82,7 +83,7 @@ WEAK char *emit_attributes_to_string(attr_list *attrs) {
 
         char strkey[64] = { 0 };
         CK_ATTRIBUTE_TYPE k = a->type;
-        snprintf(strkey, sizeof(strkey), "%lu", k);
+        snprintf(strkey, sizeof(strkey), "%"PRIu64"", k);
```

Comment:
on 32 bit systems this causes a warning:
```
a.c: In function 'main':
a.c:6:9: warning: format '%llu' expects argument of type 'long long unsigned int', but argument 2 has type 'long unsigned int' [-Wformat=]
    6 |  printf("%"PRIu64"\n", x);
      |         ^~~            ~
```

Reply:
Back to the basics, we could enforce a cast:
```
snprintf(strkey, sizeof(strkey), "%"PRIu64"", (u_int64_t)k);
```
but first, let's make sure we agree on the problem.